public/app/features/explore/Explore.tsx

```diff
@@ -1035,4 +542,61 @@ export class Explore extends React.PureComponent<ExploreProps, ExploreState> {
   }
 }
 
-export default hot(module)(Explore);
+function mapStateToProps({ explore }) {
+  const {
+    StartPage,
+    datasourceError,
+    datasourceInstance,
+    datasourceLoading,
+    datasourceMissing,
+    initialDatasource,
+    initialQueries,
+    history,
+    range,
+    showingGraph,
+    showingLogs,
+    showingStartPage,
+    showingTable,
+    supportsGraph,
+    supportsLogs,
+    supportsTable,
+  } = explore as ExploreState;
+  return {
+    StartPage,
+    datasourceError,
+    datasourceInstance,
+    datasourceLoading,
+    datasourceMissing,
+    initialDatasource,
+    initialQueries,
+    history,
+    range,
+    showingGraph,
+    showingLogs,
+    showingStartPage,
+    showingTable,
+    supportsGraph,
+    supportsLogs,
+    supportsTable,
+  };
+}
+
+const mapDispatchToProps = {
+  addQueryRow,
+  changeDatasource,
+  changeQuery,
+  clickExample,
+  clickGraphButton,
+  clickLogsButton,
+  clickTableButton,
+  highlightLogsExpression,
+  initializeExplore,
+  modifyQueries,
+  onSize: changeSize, // used by withSize HOC
```

Comment:
I would like to stop using size-me, for the react panels I am using AutoSizer from react-virtualized (https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/dashgrid/PanelChrome.tsx#L91) 
It uses render props instead and has been more reliable (withSize/sizeMe) has some strange issues that were solved by switching to AutoSizer (which I think i using a more optimized approach). But need create a tech/chore task to remove all other uses of withSize/size-me.  Think it's only DashboardGrid, explore graph, and now here. 

btw why does the explore root component care about size?

Reply:
@davkal can you elaborate a little bit more what is the relation between graph's width and interval?